test/test_settings.py

```diff
@@ -367,125 +370,67 @@ class TestSettingsEquality(object):
     A class defining tests for the standard implementation of == and != .
     """
 
-    def an_instance(self):
-        """
-        Return an instance of the class under test.  Each call to this method
-        must return a different object.  All objects returned must be equal to
-        each other.
-        """
-        overrides = {
-            h2.settings.SettingCodes.HEADER_TABLE_SIZE: 0,
-            h2.settings.SettingCodes.MAX_FRAME_SIZE: 16384,
-            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 4,
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=True, initial_values=overrides)
-
-    def another_instance(self):
-        """
-        Return an instance of the class under test.  Each call to this method
-        must return a different object.  The objects must not be equal to the
-        objects returned by an_instance.  They may or may not be equal to
-        each other (they will not be compared against each other).
-        """
-        overrides = {
-            h2.settings.SettingCodes.HEADER_TABLE_SIZE: 8080,
-            h2.settings.SettingCodes.MAX_FRAME_SIZE: 16388,
-            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: 100,
-            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: 2**16,
-        }
-        return h2.settings.Settings(client=False, initial_values=overrides)
-
-    def test_identical_eq(self):
-        """
-        An object compares equal to itself using the == operator.
-        """
-        o = self.an_instance()
-        assert (o == o)
-
-    def test_identical_ne(self):
-        """
-        An object doesn't compare not equal to itself using the != operator.
-        """
-        o = self.an_instance()
-        assert not (o != o)
-
-    def test_same_eq(self):
-        """
-        Two objects that are equal to each other compare equal to each other
-        using the == operator.
-        """
-        a = self.an_instance()
-        b = self.an_instance()
-        assert (a == b)
+    SettingsStrategy = builds(
+        h2.settings.Settings,
+        client=booleans(),
+        initial_values=fixed_dictionaries({
+            h2.settings.SettingCodes.HEADER_TABLE_SIZE: integers(0),
+            h2.settings.SettingCodes.ENABLE_PUSH: integers(0, 1),
+            h2.settings.SettingCodes.INITIAL_WINDOW_SIZE:
+                integers(0, 2**31 - 1),
+            h2.settings.SettingCodes.MAX_FRAME_SIZE:
+                integers(2**14, 2**24 - 1),
+            h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS: integers(0),
+            h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE: integers(0),
```

Comment:
Are we worried about creating values outside the valid range of some of these settings? These settings do have upper limits on their valid values.